pkg/apis/stackconfigpolicy/v1alpha1/stackconfigpolicy_types.go

```diff
@@ -0,0 +1,210 @@
+}
+
+type StackConfigPolicySpec struct {
+	ResourceSelector metav1.LabelSelector          `json:"resourceSelector,omitempty"`
+	SecureSettings   []commonv1.SecretSource       `json:"secureSettings,omitempty"`
+	Elasticsearch    ElasticsearchConfigPolicySpec `json:"elasticsearch,omitempty"`
+}
+
+type ElasticsearchConfigPolicySpec struct {
+	// ClusterSettings holds the Elasticsearch cluster settings (/_cluster/settings)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	ClusterSettings *commonv1.Config `json:"clusterSettings,omitempty"`
+	// SnapshotRepositories holds the Snapshot Repositories settings (/_snapshot)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	SnapshotRepositories *commonv1.Config `json:"snapshotRepositories,omitempty"`
+	// SnapshotLifecyclePolicies holds the Snapshot Lifecycle Policies settings (/_slm/policy)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	SnapshotLifecyclePolicies *commonv1.Config `json:"snapshotLifecyclePolicies,omitempty"`
+	// SecurityRoleMappings holds the Role Mappings settings (/_security/role_mapping)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// SecurityRoleMappings *commonv1.Config `json:"securityRoleMapping,omitempty"`
+	// AutoscalingPolicies holds the Autoscaling Policies settings (/_autoscaling/policy)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// AutoscalingPolicies *commonv1.Config `json:"autoscalingPolicies,omitempty"`
+	// IndexLifecyclePolicies holds the Index Lifecycle policies settings (/_ilm/policy)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// IndexLifecyclePolicies *commonv1.Config `json:"indexLifecyclePolicies,omitempty"`
+	// IngestPipelines holds the Ingest Pipelines settings (/_ingest/pipeline)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	// IngestPipelines *commonv1.Config `json:"ingestPipelines,omitempty"`
+	// IndexTemplates holds the Index and Component Templates settings
+	// IndexTemplates *IndexTemplates `json:"indexTemplates,omitempty"`
+}
+
+/*type IndexTemplates struct {
+	// ComponentTemplates holds the Component Templates settings (/_component_template)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	ComponentTemplates *commonv1.Config `json:"componentTemplates,omitempty"`
+	// ComposableIndexTemplates holds the Index Templates settings (/_index_template)
+	// +kubebuilder:pruning:PreserveUnknownFields
+	ComposableIndexTemplates *commonv1.Config `json:"composableIndexTemplates,omitempty"`
+}*/
+
+type StackConfigPolicyStatus struct {
+	// ResourcesStatuses holds the status for each resource to be configured.
+	ResourcesStatuses map[string]ResourcePolicyStatus `json:"resourcesStatuses"`
+	// Resources is the number of resources to be configured.
+	Resources int `json:"resources,omitempty"`
+	// Ready is the number of resources successfully configured.
+	Ready int `json:"ready,omitempty"`
+	// Errors is the number of resources which have an incorrect configuration
+	Error int `json:"errors,omitempty"`
```

Comment:
Is this supposed to be `Error` or `Errors`?  The comment and json tag match, but the name does not.

Reply:
`Errors`, well seen.